src/lightning/fabric/fabric.py

```diff
@@ -678,8 +683,21 @@ def save(self, path: Union[str, Path], state: Dict[str, Union[nn.Module, Optimiz
             path: A path to where the file(s) should be saved
             state: A dictionary with contents to be saved. If the dict contains modules or optimizers, their
                 state-dict will be retrieved and converted automatically.
+            filter: An optional dictionary containing filter callables that return a boolean indicating whether the
+                given item should be saved (``True``) or filtered out (``False``). Each filter key should match a
+                state key, where its filter will be applied to the ``state_dict`` generated.
         """
-        self._strategy.save_checkpoint(path=path, state=_unwrap_objects(state))
+        if filter is not None:
+            if not isinstance(filter, dict):
```

Comment:
if it's just a plain callable, should this one automatically be applied to the model only?